### DISCUSSION TITLE
Ikea TRADFRI shortcut button all trigger an Aqara Plug

### DISCUSSION THREAD

**Discussion by Nanunan** [Koenkk/zigbee2mqtt#17244] [Devices]
Hello everyone,

I installed 3 Ikea TRADFRI shortcut buttons into my Zigbee2MQTT. All three have a different automation that I created in home Assistant.

One of them triggers a aqara smart plug. Heres my problem: The other 2 also trigger the smart plug. As seen in the picture below, the trigger from the incorrect shortcut buttons appears different. Can someone help me with that or explain what is happening?

![ZigbeeDevices2](https://user-images.githubusercontent.com/38922933/229871683-56013add-affa-4d91-92cf-130b5d66b5b4.PNG) 

From buttom to top: First two triggered by the correct button, in the middle triggered in home assistant, and the top one is triggerd by the wrong ikea shortcut button.

![ZigbeeDevices](https://user-images.githubusercontent.com/38922933/229871809-8034dac9-fb55-43b1-9d65-c9a28f46cccb.PNG)

Devices in Z2M

**Comment by robertwigley** [comment]
Did you ever get to the bottom of this? I have 5 IKEA TRADFRI Shortcut Buttons (E1812), which all seem to have invisibly bound themselves to my Aqara Smart Wall Switch H1 EU (With Neutral) Double Rocker (WS-EUK04) switches and are switching them on whenever the IKEA buttons are pressed. I simply can't see how to remove the bindings, as they are not visible anywhere in Zigbee2MQTT.

Tagging @Koenkk to see if he can offer any advice, or whether this might be a bug that needs investigating.

**Comment by Nanunan** [comment]
Hello! 
No, I didnt. Couldnt find anything that helps. 
It does not happen every time I press the IKEA button, but I couldnt find out what made the difference to trigger the Aqara plug.

**Reply by robertwigley** [reply]
For me it happens every time and is consistently reproduceable. The buttons are only triggering the Aqara switch they are routed through it. If I remove/reset and re-pair it happens again immediately (if paired through a switch). IMO it has to be binding itself to the switch, but the binding is not visible anywhere and I can't see how to unbind it becuase it's not visible. I even tried adding a binding and removing it, but that makes no difference either.

I have about 25 of the Aqara switches (I fitted them throughout my house) in a combination of the double rocker and single rocker (WS-EUK03) model, so they form the backbone of my Zigbee mesh and I can't very much avoid the IKEA buttons routing through them. When connected directly to the coordinator, the problem does not seem to occur, although I need to do more testing on that.

It's either an incompatiblity or a bug, but I am not sure which at the moment.

My Zigbee knowledge is basic to moderate. I am learning slowly, but I don't really have a full grasp on how endpoints and bindings work yet.

**Comment by robertwigley** [comment]
@Nanunan If you load the Zigbee network map do your buttons get routed through the Aqara plug? As the Zigbee mesh can reconfigure and change itself, is there a correlation between when they are routed through the plug to when the problem occurs and vice-versa?

**Comment by robertwigley** [comment]
I can see the following in the logs:

```
debug 2023-06-20 14:33:15Received Zigbee message from 'Status Button', type 'commandOn', cluster 'genOnOff', data '{}' from endpoint 1 with groupID null
info 2023-06-20 14:33:15MQTT publish: topic 'zigbee2mqtt/Status Button', payload '{"action":"on","battery":90,"linkquality":255,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null}'
info 2023-06-20 14:33:15MQTT publish: topic 'zigbee2mqtt/Status Button', payload '{"action":"","battery":90,"linkquality":255,"update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null}'
info 2023-06-20 14:33:15MQTT publish: topic 'zigbee2mqtt/Status Button/action', payload 'on'
debug 2023-06-20 14:33:15Received Zigbee message from 'Office Light Switch', type 'attributeReport', cluster 'genOnOff', data '{"245":124340839,"onOff":1}' from endpoint 1 with groupID null
info 2023-06-20 14:33:15MQTT publish: topic 'zigbee2mqtt/Office Light Switch', payload '{"action":null,"consumption":4.45,"current":0,"device_temperature":34,"energy":4.45,"flip_indicator_light":"ON","led_disabled_night":false,"linkquality":255,"operation_mode":"decoupled","power":0.66,"power_outage_count":22,"power_outage_memory":true,"state":"ON","update":{"installed_version":2839,"latest_version":2839,"state":"idle"},"update_available":false,"voltage":237.46}'
```

So clearly the Light Switch is being sent an ON payload at the same time that the button is being pressed and sending it's ON playload.

**Comment by Nanunan** [comment]
When I open the map, I can confirm that the IKEA button that triggers the aqara is routed trough it. Is it possible to rerout it?
I have only basic knowledge in Z2M, as I do most of the automations in Home assistant.

**Reply by robertwigley** [reply]
I am no expert with Zigbee or Z2M either. The only way I know to reroute it, is to remove the device and re-pair it, forcing it through a different router or direct to the coordinator using the Permit Join dropdown to select which you want to allow joining though. I have removed one of my buttons and forced it to join via the coordinator, which appears to have fixed the problem on that one for now. However, the problem is that Zigbee will reconfigure itself as it's self healing (I don't know exactly what will trigger it to do this) and may reroute things differently when it does. So how long this will last, before it ends up connecting and routing itself back trhough one of the Aqara switches, I don't know. I am hoping that maybe if it is force joined via the coordinator, it won't happen when it reroutes itself. Maybe it only happens when it joins though one of them. Only time will tell...

**Reply by Nanunan** [reply]
Definetly keep me updated on that! If it works, I will do the same.

**Reply by robertwigley** [reply]
Re-pairing it through the coordinator did not work. It rerouted itself to another Aqara switch and the problem returned. I cannot find a solution to this, so I have given up on using the IKEA buttons and switched to using the Aqara buttons instead, which do not have this problem.

**Comment by tomsdk** [comment]
This is a well known issue with aqara. There is a downgrade method for their smart switches to work around the issue until fixed.

**Reply by robertwigley** [reply]
@tomsdk Please can you share what this is and how to do it? I have spent so long trying to resolve this and searching for an answer that I just haven't been able to find, that I gave up on the IKEA buttons. However, if there is a way to work around this, it would be great, as I don't really want to ditch the 5 buttons I have.

**Reply by Nanunan** [reply]
Second this. Also I didnt find any info about the known issue. Can you tell us about the work around?

**Reply by jeppester** [reply]
I believe it's this comment:
https://github.com/Koenkk/zigbee2mqtt/issues/13903#issuecomment-1244720072

**Comment by Koenkk** [comment]
There are some tips in this issue, maybe it helps: https://github.com/Koenkk/zigbee2mqtt/issues/16918

**Reply by robertwigley** [reply]
Thank you @Koenkk. That's the exact same issue. I already tried everything suggested, checking for bindings, groups, default groups, but with everything I tried, the problem just keeps coming back whenever the button becomes a child of one of the Aqara switches.

I even removed the button from Zigbee2MQTT and the problem remains, so I am not sure this is even a Zigbee2MQTT problem. It's somehow creating a local binding to the Aqara switch whenever it's a child of it, that's not visible in Zigbee2MQTT. AFAIK there isn't a way to lock a Zigbee device to a non-Aqara router, so joining to another router is only a short lived fix, as they automatically reroute themselves every now and then and when hooking onto an Aqara switch the problem returns.

I did recently note, while swapping over to the Aqara buttons, that the actions sent by them, which do not have this problem, are `single`, `double`, `hold` and `release`. Whereas the IKEA buttons send actions `on`, `off`, `brightness_move_up` and `brightness_stop`. A single-click is `on`, which is what's triggering the relay it's a child of to switch on. A double-click is `off`, which does the inverse and switches the relay off. It seems like because the actions are `on` and `off`, the Aqara switch is interpreting that as a command to do that in the relay. I'm not sure how much this extra information helps with this issue, or if this can even be fixed and is simply an unresolvable incompatiblity between devices. One, or both devices can't be following Zigbee standards correctly I suspect. Probably the Aqara ones.

**Reply by Koenkk** [reply]
Can you make a sniff both when the device is a directly child (Xiaomi toggles) and when it's not a child (Xiaomi doesn't toggle)? Please provide the sniff here: https://github.com/Koenkk/zigbee2mqtt/issues/16918 (https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html)

**Reply by robertwigley** [reply]
I'd love to provide this info, but, unfortunately, I don't have the required hardware and I literally finished removing all my IKEA buttons from my network today. I can re-pair one to test and get more information and am happy to do so, but unless there is a way to sniff the info without spending money on hardware I don't have, then I can't help with this unfortunately. I just have a regular setup with a Conbee II stick running on a RPi4.

**Reply by quidrick** [reply]
@Koenkk I have this issue and a spare ZigStar v4 (CC2652P). Is there any way to do a sniff with this stick?

**Reply by Koenkk** [reply]
https://www.ti.com/tool/download/PACKET-SNIFFER-2/1.10.0 should include a firmware but I have no clue if it is going to work on the ZigStar v4.

**Comment by adynis** [comment]
I was so happy when I purchased ikea symfonisk remote gen 2 ... and I'm so unhappy now to see that ther's this issue :( 
Lots of Aqara EUK04 in house ... Ikea remote, when I press the middle button it triggers on/off the "parent" switch .... :( 
Hoping for a future solution ....

**Comment by carlos-otero** [comment]
Hi, I have the same issue here.

set up RODRET and STYRBAR switches from Ikea in Zigbee2mqtt and without any automation set, it turns on and off an Aqara wall switch. After setting an automation, it keeps turning that wall switch on and off. Ant ideas? Thank you!

**Comment by viharm** [comment]
Yes, I have the same issue with a RODRET remote and the double-rocker Aqara wall switch (with neutral). I can confirm that in the map, the remote connects via the affected Aqara wall switch, even if I force to join directly to the coordinator. This is confirmed in the [Z2M guide](https://www.zigbee2mqtt.io/guide/usage/pairing_devices.html#pairing-via-a-specific-device), which states...

> Pairing via a specific device
> 
> To the right of Permit join (All) button is an arrow pointing down. This will allow you to pick any router to join from. That is not a guarantee the router is chosen. This can be useful especially for Aqara devices or a switch to a light bulb.

**Comment by Pospitch** [comment]
So I have exactly same issue and it's really annoying, especially when switching off light switch off my PC.. I see this issue is ongoing for years now. Any idea why it hasn't been fixed yet? Did anyone tried to contact manufacturer or devs? I know there is some workaround, but I only have Aeotect Hub and Smartthings app, so I can't downgrade anything.

**Comment by dominicusmento** [comment]
Same problem here... except I've decupled (it suits me) my WS-EUK03 switches so they only change relay states when ikea btns are pressed.. Annoying still as they produce the sound when state is changing..

**Comment by jeppester** [comment]
Same issue here, RODRET toggles the Aqara Smart Plug (SP-EUC01) it's routed through.
Does anyone know if this is a RODRET or Aqara issue? I want to replace the device causing the problem.

**Reply by tomsdk** [reply]
You can downgrade the firmware on the aqara to a version without the flaw.

**Reply by jeppester** [reply]
Thank you, I'll see if I can make it work.

In case others come looking I believe these are the instructions to follow:
https://github.com/Koenkk/zigbee2mqtt/issues/13903#issuecomment-1244720072

**Reply by jeppester** [reply]
I downgraded all my plugs and the problem is gone for now :tada: 

And I won't buy anything from aqara again that has a switch

**Reply by tomsdk** [reply]
Great! Fun fact - if you buy that product today, I think it has been changed to an end device, so routing is not possible. Problem solved :)

**Reply by jeppester** [reply]
It was mostly the T1 (with neutral) switch module that would be relevant for me.
Unfortunately only the no-neutral version has a notice about not functioning as a router. :smile:

**Comment by dominicusmento** [comment]
Aqara, I have the same issue with Ikea and Sonoff buttons

**Reply by jeppester** [reply]
Did you try the suggestion here?:
https://github.com/Koenkk/zigbee2mqtt/issues/13903#issuecomment-1244720072
